### ISSUE TITLE
Convert OCI Build Task to Prototype

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#6584]
### TLDR

- Simplifying the ability for Concourse to use a container image to build other container images utilizing Prototypes as opposed to the OCI Build Task. 

### Problem

- Currently, the [`oci-build-task`](https://hub.docker.com/r/vito/oci-build-task) is a container image that is meant to be used in a Concourse task to build other container images. 
- The process to build an OCI image from its Dockerfile and push that image to a registry involves quite a lot of steps and can be verbose and complicated

### Proposed Solution

- Utilize Prototypes instead! Given that prototypes is built off the back of resources, it should be an easier abstraction to reason about.

### Outcomes

- OCI images become easier to build and push
- Users are provided the same flexibility and power, but in an easier to manage format 

### Current Status

- Blocked by ongoing initial prototypes work

### References

https://github.com/concourse/concourse/pull/6540 - Add support for prototypes

See: [Reinventing Resource Types](https://blog.concourse-ci.org/reinventing-resource-types/)
